packages/client/src/generation/TSClient/Model.ts

```diff
@@ -282,54 +286,67 @@ export type ${getAggregateGetName(model.name)}<T extends ${getAggregateArgsName(
 
     return ifExtensions(
       () => {
-        return `export type ${
-          model.name
-        }Payload<ExtArgs extends runtime.Types.Extensions.Args = runtime.Types.Extensions.DefaultArgs> = {
-${indent(
-  `objects: {
-${indent(
-  model.fields
-    .filter((f) => f.kind === 'object')
-    .map((field) => new ModelOutputField(this.dmmf, field, false).toTS())
-    .join('\n'),
-  TAB_SIZE,
-)}
-}`,
-  TAB_SIZE,
-)}
-${indent(
-  `scalars: runtime.Types.Extensions.GetResult<{
-${indent(
-  model.fields
-    .filter((f) => f.kind === 'scalar' || f.kind === 'enum')
-    .map((field) => new ModelOutputField(this.dmmf, field, !this.dmmf.typeMap[field.type]).toTS())
-    .join('\n'),
-  TAB_SIZE,
-)}
-}, ExtArgs['result']['${lowerCase(model.name)}']>`,
-  TAB_SIZE,
-)}
-}
+        const objects = ts.objectType()
+        const scalars = ts.objectType()
+        const composites = ts.objectType()
+
+        for (const field of model.fields) {
+          if (field.kind === 'object') {
+            if (this.dmmf.typeMap[field.type]) {
+              composites.add(buildModelOutputProperty(field, this.dmmf))
+            } else {
+              objects.add(buildModelOutputProperty(field, this.dmmf))
```

Comment:
Can this happen?

Reply:
Yes. `typeMap` is little bit of a misleading name, but it lists all comosites types. So, if type is listed in `typeMap` it is a composite, if it is not it's just a regular relation.